File: python/SivitasAkademik.py
# Membuat kelas baru bernama SivitasAkademik, class ini merupakan child class dari Human Class
from Human import Human

class SivitasAkademik(Human):
    
    # Inisialisasi variabel untuk setiap objek dari kelas
    asal_univ = ""  # Membuat variabel untuk asal universitas
    email_edu = ""  # Membuat variabel untuk email pendidikan
    
    # Konstruktor untuk inisialisasi variabel pada objek dari kelas SivitasAkademik
    def __init__(self, NIK = "", name = "", gender = "", asal_univ = "", email_edu = ""):
        super().__init__(NIK, name, gender)
        self.asal_univ = asal_univ  # Menginisialisasi variabel asal_univ
        self.email_edu = email_edu  # Menginisialisasi variabel email_edu
        
    # Getter dan Setter untuk Asal Universitas
    def get_Asal_univ(self):
        return self.asal_univ
    
    def set_Asal_univ(self, asal_univ):
        self.asal_univ = asal_univ
        
    # Getter dan Setter untuk Email_edu
    def get_Email_edu(self):
        return self.email_edu
    
    def set_Email_edu(self, email_edu):
        self.email_edu = email_edu

File: cpp/SivitasAkademik.cpp
class SivitasAkademik :  public Human{
    
    /*
    Saya Talitha Syahla NIM 2101330 mengerjakan Soal Latihan 4 
    dalam mata kuliah Desain Pemrograman Berorientasi Objek 
    untuk keberkahanNya maka saya tidak melakukan kecurangan 
    seperti yang telah dispesifikasikan. Aamiin.
    */

    private:
        string asal_univ;
        string email_edu;

    public:
    // Constructor
        SivitasAkademik(){
            this->asal_univ = "";
            this->email_edu = "";
        }

        SivitasAkademik(string NIK, string name, string gender, string asal_univ, string email_edu) : Human(NIK, name, gender){
            this->asal_univ = asal_univ;
            this->email_edu = email_edu;
        }

    // Setter dan Getter

        // Untuk Asal Universitas
        string get_Asal_univ(){
            return this->asal_univ;
        }

        void set_Asal_univ(string asal_univ){
            this->asal_univ = asal_univ;
        }

        // Untuk Email_edu
        string get_Email_edu(){
            return this->email_edu;
        }

        void set_Email_edu(string email_edu){
            this->email_edu = email_edu;
        }

        // Destruktor
        ~SivitasAkademik(){
        }
    
};